### PR TITLE
Dmi date fix

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

File: src/mainboard/pcengines/apu2/romstage.c [hunk 1]
```diff
@@ -208,9 +209,14 @@ void cache_as_ram_main(unsigned long bist, unsigned long cpu_init_detectedx)
...
 	if(scon){
+		char tmp[9];
+		strncpy(tmp,   coreboot_dmi_date+6, 4);
+		strncpy(tmp+6, coreboot_dmi_date+3, 2);
+		strncpy(tmp+4, coreboot_dmi_date,   2);
+		tmp[8] = '\0';
 		printk(BIOS_ALERT, CONFIG_MAINBOARD_SMBIOS_MANUFACTURER " "
 		                   CONFIG_MAINBOARD_PART_NUMBER "\n");
-		printk(BIOS_ALERT, "coreboot build %s\n", COREBOOT_DMI_DATE);
+		printk(BIOS_ALERT, "coreboot build %s\n", tmp);
```
Comment (pietrushnic) [inline src/mainboard/pcengines/apu2/romstage.c]: @miczyg1 please confirm if I understand that correctly, `COREBOOT_DMI_DATE` was `%Y/%m/%d` and now is `%m/%d/%Y`, tmp is `%Y/%d/%m? Why? Please add correct comment in code to.

Comment (krystian-hebel) [reply src/mainboard/pcengines/apu2/romstage.c]: @pietrushnic tmp is `%Y%d%m`, without slashes.

**Comment by michaelsteinmann** [comment]
Shouldn't that be YMD instead of YDM?
According to https://en.wikipedia.org/wiki/Date_format_by_country *"**ydm"*
is only used in Kazakh in Kazakhstan (Russian in Kazakhstan has DMY),
Latvian in Latvia



Am Di., 23. Okt. 2018 um 10:55 Uhr schrieb krystian-hebel <
notifications@github.com>:

File: src/mainboard/pcengines/apu2/romstage.c (hunk 1 above)
Comment (miczyg1) [reply src/mainboard/pcengines/apu2/romstage.c]: @krystian-hebel slashes are one thing.
@pietrushnic added correct comment. `tmp` was set to yyyymmdd incorrectly (I have mistaken the notations), should be yyyyddmm. Already fixed

**Comment by miczyg1** [comment]
@michaelsteinmann That is what I thought in the first place too, however when I looked on mainline releases date format, it was YYYYDDMM. So I wanted to align to it

**Comment by miczyg1** [comment]
I have flashed and checked the very first binary we have for apu2 (from 12th of September 2016) and the date was 20161209. So it was in Kazakhstan format from the beginning.

**Comment by michaelsteinmann** [comment]
Where is the date stored in this stupid format? dmidecode shows YMD format
correct:

BIOS Information
        Vendor: coreboot
        Version: v4.0.17
        Release Date: 20180511


Am Di., 23. Okt. 2018 um 11:38 Uhr schrieb Michał Żygowski <
notifications@github.com>:

**Comment by miczyg1** [comment]
@michaelsteinmann according to SMBIOS specification YMD format is wrong in dmidecode. Have a look at the comment in this issue: https://github.com/pcengines/apu2-documentation/issues/117#issuecomment-418794655

The date is generated during build and is included in header files during build and as a string after linking.

**Pull request merged**